### PR TITLE
Rust 2018

### PR DESCRIPTION
**Pull request by daboross**
Updates this project to rust 2018 edition.

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -1,15 +1,12 @@
 #![recursion_limit = "128"]
-extern crate fern;
-#[macro_use]
-extern crate log;
-extern crate screeps;
 #[macro_use]
 extern crate stdweb;
```
Comment (ASalvail) [inline src/main.rs]: Following https://doc.rust-lang.org/stable/edition-guide/rust-2018/macros/macro-changes.html

This should be switched to `use stdweb::js` .

Comment (daboross) [reply src/main.rs]: Right, guess this should probably change too!

It currently requires importing a helper macro too, but I've filed an issue for that: https://github.com/koute/stdweb/issues/347

File: src/main.rs [hunk 2]
```diff
@@ -119,7 +113,7 @@ fn game_loop() {
     info!("done! cpu: {}", screeps::game::cpu::get_used())
 }
 
-fn cleanup_memory() -> Result<(), Box<::std::error::Error>> {
+fn cleanup_memory() -> Result<(), Box<dyn ::std::error::Error>> {
```
Comment (ASalvail) [inline src/main.rs]: I think you don't need the `::` in front of `std` anymore with rust 2018.

File: src/main.rs [hunk 3]
```diff
@@ -1,16 +1,10 @@
-#![recursion_limit = "128"]
```
Comment (ASalvail) [inline src/main.rs]: Any reason this was added? Just for the `js` macro? Has it been fixed?

Comment (daboross) [reply src/main.rs]: Errr I think I just tried removing it and forgot to re-add it in when it worked. I'll double check.

File: src/main.rs (hunk 2 above)
Comment (daboross) [reply src/main.rs]: right!

File: src/main.rs (hunk 3 above)
Comment (daboross) [reply src/main.rs]: I think stdweb 0.4.11's change to a proc macro might have made it unnecessary?

I'll stick it in a different PR though

**Pull request merged**